`cuSparseTest.test_coo_sorted_indices_gpu_lowerings` test fails when running on a CPU-only machine

Please:

- [x] Check for duplicate issues.
- [x] Provide a complete example of how to reproduce the bug, wrapped in triple backticks like this:

1. Install jax v0.3.5 and CUDA-enabled jaxlib v0.3.0. 
2. Run the test suite.
3. Observe the `cuSparseTest.test_coo_sorted_indices_gpu_lowerings` test to fail:
```
=================================== FAILURES ===================================
______________ cuSparseTest.test_coo_sorted_indices_gpu_lowerings ______________
[gw0] linux -- Python 3.9.11 /nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/bin/python3.9

self = <sparse_test.cuSparseTest testMethod=test_coo_sorted_indices_gpu_lowerings>

    @unittest.skipIf(not GPU_LOWERING_ENABLED, "test requires cusparse/hipsparse")
    @jtu.skip_on_devices("rocm")  # TODO(rocm): see SWDEV-328107
    def test_coo_sorted_indices_gpu_lowerings(self):
      dtype = jnp.float32

      mat = jnp.arange(12, dtype=dtype).reshape(4, 3)

      mat_rows_sorted = sparse.COO.fromdense(mat)
      self.assertTrue(mat_rows_sorted._rows_sorted)
      self.assertFalse(mat_rows_sorted._cols_sorted)

      mat_cols_sorted = sparse.COO.fromdense(mat.T).T
      self.assertFalse(mat_cols_sorted._rows_sorted)
      self.assertTrue(mat_cols_sorted._cols_sorted)

      mat_unsorted = sparse.COO(mat_rows_sorted._bufs, shape=mat_rows_sorted.shape)
      self.assertFalse(mat_unsorted._rows_sorted)
      self.assertFalse(mat_unsorted._cols_sorted)

      self.assertArraysEqual(mat, mat_rows_sorted._sort_rows().todense())
      self.assertArraysEqual(mat, mat_cols_sorted._sort_rows().todense())
      self.assertArraysEqual(mat, mat_unsorted._sort_rows().todense())

      todense = jit(sparse.coo_todense)
      with self.assertNoWarnings():
        dense_rows_sorted = todense(mat_rows_sorted)
        dense_cols_sorted = todense(mat_cols_sorted)
        dense_unsorted = todense(mat_unsorted._sort_rows())
      with self.assertWarnsRegex(sparse.CuSparseEfficiencyWarning, "coo_todense GPU lowering requires matrices with sorted rows.*"):
>       dense_unsorted_fallback = todense(mat_unsorted)
E       AssertionError: CuSparseEfficiencyWarning not triggered

tests/sparse_test.py:473: AssertionError
=========================== short test summary info ============================
FAILED tests/sparse_test.py::cuSparseTest::test_coo_sorted_indices_gpu_lowerings
========== 1 failed, 14145 passed, 1487 skipped in 165.49s (0:02:45) ===========
error: builder for '/nix/store/bk64c49429aj73wkz4dijlf15h0l2v9r-python3.9-jax-0.3.5.drv' failed with exit code 1;
       last 10 log lines:
       >         dense_cols_sorted = todense(mat_cols_sorted)
       >         dense_unsorted = todense(mat_unsorted._sort_rows())
       >       with self.assertWarnsRegex(sparse.CuSparseEfficiencyWarning, "coo_todense GPU lowering requires matrices with sorted rows.*"):
       > >       dense_unsorted_fallback = todense(mat_unsorted)
       > E       AssertionError: CuSparseEfficiencyWarning not triggered
       >
       > tests/sparse_test.py:473: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/sparse_test.py::cuSparseTest::test_coo_sorted_indices_gpu_lowerings
       > ========== 1 failed, 14145 passed, 1487 skipped in 165.49s (0:02:45) ===========
```
- [x] If applicable, include full error messages/tracebacks.

Why isn't this test skipped when running on a machine without access to a GPU?

--- a/tests/sparse_test.py
+++ b/tests/sparse_test.py
@@ -442,6 +442,7 @@ class cuSparseTest(jtu.JaxTestCase):
     self.assertArraysEqual(mat.todense(), mat_resorted.todense())
 
   @unittest.skipIf(not GPU_LOWERING_ENABLED, "test requires cusparse/hipsparse")
+  @unittest.skipIf(jtu.device_under_test() != "gpu", "test requires GPU")
   @jtu.skip_on_devices("rocm")  # TODO(rocm): see SWDEV-328107
   def test_coo_sorted_indices_gpu_lowerings(self):
     dtype = jnp.float32

